### PR TITLE
Add not_empty_string test

### LINKED ISSUE #632: not_blank test
### Describe the feature
dbt has the `not_null` test to monitor for null values, but in certain datasets blank values are also problematic. However, there is no `not_blank` test that I found when searching utils or dbt-expectations.
### Describe alternatives you've considered
Creating a custom test that errors if there are blanks.
### Additional context
Is this feature database-specific? Which database(s) is/are relevant? Please include any other relevant context here.
I think blank entries are a possibility on all databases
### Who will this benefit?
What kind of use case will this feature be useful for? Please be specific and provide examples, this will help us prioritize properly.
Anyone needing to check data integrity to ensure there are no blank values in their dataset
### Are you interested in contributing this feature?

Yes plz 🙏🏻

### PR DESCRIPTION
**Pull request by epapineau**
This is a:
- [ ] documentation update
- [ ] bug fix with no breaking changes
- [x] new functionality
- [ ] a breaking change

All pull requests from community contributors should target the `main` branch (default).

## Description & motivation

Resolves #632 
## Checklist
- [x] I have verified that these changes work locally on the following warehouses (Note: it's okay if you do not have access to all warehouses, this helps us understand what has been covered)
    - [x] BigQuery
    - [x] Postgres
    - [x] Redshift
    - [x] Snowflake
- [x] I followed guidelines to ensure that my changes will work on "non-core" adapters by:
    - [x] dispatching any new macro(s) so non-core adapters can also use them (e.g. [the `star()` source](https://github.com/dbt-labs/dbt-utils/blob/main/macros/sql/star.sql))
    - [ ] using the `limit_zero()` macro in place of the literal string: `limit 0`
    - [ ] using `dbt_utils.type_*` macros instead of explicit datatypes (e.g. `dbt_utils.type_timestamp()` instead of `TIMESTAMP`
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)
- [x] I have added an entry to CHANGELOG.md

### CODE REVIEW

**Comment by epapineau** [comment]
We discussed on the original issue also adding a `not_empty_array` test, but I need to put more thought into what that would look like from a testing perspective (see [core issue 2365](https://github.com/dbt-labs/dbt-core/issues/2365)). I wanted to get this out for review now though.

File: integration_tests/models/sql/schema.yml [hunk 1]
```diff
@@ -182,3 +182,29 @@ models:
...
+
+  - name: test_not_empty_string_failing
+    columns:
+      - name: string
+        tests:
+          - dbt_utils.not_empty_string:
+              trim_whitespace: True
+              config:
+                severity: error
+                error_if: "<1"
```
Comment (joellabes) [inline integration_tests/models/sql/schema.yml:194]: Should these be greater than signs?

Comment (joellabes) [reply integration_tests/models/sql/schema.yml:194]: No, they shouldn't! This is a _very_ clever way to force the tests to check the negative and positive case. 

I really like this 🧠

File: integration_tests/models/sql/schema.yml [hunk 2]
```diff
@@ -182,3 +182,29 @@ models:
...
+                warn_if: "<0"
+
+  - name: test_not_empty_string_passing
+    columns:
+      - name: string
+        tests:
+          - dbt_utils.not_empty_string:
+              trim_whitespace: True
+          - dbt_utils.not_empty_string:
+              trim_whitespace: False
```
Comment (joellabes) [inline integration_tests/models/sql/schema.yml:210]: If you can run the same test on the same column with two different configs, is one of the configs redundant? This feels odd to me

**Review by joellabes — COMMENTED** [review]
I like where this is going! Thank you for your patience 🙏 a couple of comments/questions where I'm not sure that I'm understanding it correctly, but otherwise let's get this merged

File: integration_tests/models/sql/schema.yml (hunk 1 above)
Comment (epapineau) [reply integration_tests/models/sql/schema.yml:194]: All credit goes to @dbeatty10 on this approach, but it feels good knowing that we're testing both the pass and fails.

File: integration_tests/models/sql/schema.yml (hunk 2 above)
Comment (epapineau) [reply integration_tests/models/sql/schema.yml:210]: I totally see your point here. I was trying to also test the `trim_whitespace` option. I've removed the redundant ones for now, but let me know if you have another suggestion for testing the option.

Comment (joellabes) [reply integration_tests/models/sql/schema.yml]: I _think_ you just need to test two different columns? One with whitespace that needs trimmed to pass, and one that doesn't.

Comment (epapineau) [reply integration_tests/models/sql/schema.yml]: @joellabes brilliant as always~✨ Added a `string_trim_whitespace_false` column to `test_not_empty_string_passing` that would cause the test to fail if we weren't properly trimming whitespace :)

**Review by joellabes — COMMENTED** [review]
@epapineau this looks great! just checking if you meant to change the postgres env vars?

File: integration_tests/.env/postgres.env [hunk 3]
```diff
@@ -1,5 +1,5 @@
 POSTGRES_TEST_HOST=localhost
-POSTGRES_TEST_USER=root
+POSTGRES_TEST_USER=elizepapineau
```
[the remark this answers is not in the document]
Comment (epapineau) [reply integration_tests/.env/postgres.env]: No 🤦🏻

[the remark this answers is not in the document]
Comment (epapineau) [reply integration_tests/.env/postgres.env]: Fixed, thanks for pointing out~

**Review by joellabes — APPROVED** [review]
